plugins/modules/iosxr_config.py

```diff
@@ -451,6 +458,17 @@ def main():
         supports_check_mode=True,
     )
 
+    connection = get_connection(module)
+    os_version = get_os_version(connection)
+    if os_version > "7.2":
```

Comment:
Instead of having branch AnsibleModule() twice, can we apply the logic here - https://github.com/ansible-collections/cisco.iosxr/blob/c31d6fa26c0a7d9bbda194b816b93a18b3416bfd/plugins/modules/iosxr_config.py#L270 to set DEFAULT_COMMIT_COMMENT to none if the version doesn't support it?

Reply:
But without module object we can not get os version

Reply:
Setting `None` at global level might not work as the version information is available after module is instantiated.

Instead we can remove the default value `DEFAULT_COMMIT_COMMENT ` for `comment` option and explicitly set it after module object is instantiated based on the version as required. For example 
```
    os_version = get_os_version()
    if os_version > "7.2":
        if module["comment"]:
            msg = "value of comment option '%s' is ignored as it in not supported by os version '%s'" % (module["comment"], os_version)
            warnings.append(msg)
    else:
        # for backward compatibility
        if not module["comment"]:
            module["comment"] = DEFAULT_COMMIT_COMMENT
```

Reply:
with this code we always setting comment to None for iosxr version > 7.2 .this is also not expected right. i mean what if they specified comment value explicitly in the playbook in that case the above code will fail.

Reply:
os_version = get_os_version()
    if version.parse(os_version) > version.parse("7.2"):
        if module["comment"] and module["comment"] =="DEFAULT_COMMIT_COMMENT":
            msg = "value of comment option '%s' is ignored as it in not supported by os version '%s'" % (module["comment"], os_version)
            warnings.append(msg)
            module["comment"] = None

Reply:
Based on the issue discussed  it seems like passing value of comment option (default or explicitly set by user) itself is is not supported for version 7.2 and above.

Reply:
What happens if user passes comment value from task with iosxr 7.2 version?

Reply:
oh okay